mlrun/__main__.py

```diff
@@ -753,16 +753,51 @@ def project(
         proj.sync_functions(save=True)
 
 
+def validate_kind(ctx, param, value):
+    possible_kinds = RuntimeKinds.runtime_with_handlers()
+    if value != '' and value not in possible_kinds:
+        raise click.BadParameter(
+            f'kind must be one of {possible_kinds}', ctx=ctx, param=param
+        )
+    return value
+
+
 @main.command()
-@click.argument('kind', type=str, default='', required=False)
+@click.argument('kind', callback=validate_kind, default='', required=False)
 @click.argument('object_id', metavar='id', type=str, default='', required=False)
-@click.option('--api', help='api and db service url')
+@click.option('--api', help='api service url')
 @click.option('--label-selector', '-ls', default='', help='label selector')
 @click.option(
     '--force', '-f', is_flag=True, help='clean resources in transient states as well'
 )
-def clean(kind, object_id, api, label_selector, force):
-    """Clean runtime resources"""
+@click.option(
+    '--grace-period',
+    '-gp',
+    type=int,
+    default=mlconf.runtime_resources_deletion_grace_period,
+    help="the grace period (in seconds) that will be given to runtime resources (after they're in stable state) "
+    "before cleaning them. Irrelevant when using with --force",
+    show_default=True,
+)
+def clean(kind, object_id, api, label_selector, force, grace_period):
+    """
+    Clean jobs resources
+
+    \b
+    Examples:
+
+        \b
+        # Clean resources for all jobs of all runtimes
+        mlrun clean
+
+        \b
+        # Clean resources for all jobs of specific kind (job)
```

Comment:
```suggestion
        # Clean resources for all runs of a specific kind (e.g. job)
```
Not sure about the terminology here, what is the instance of a thing called if it's not a job?